models/archive-exhibition.php

```diff
@@ -327,6 +366,81 @@ public function years() {
         return $choices;
     }
 
+    /**
+     * Reorder main exhibitions to top of other exhibitions with the same dates
+     *
+     * @param array $items Array of WP_Post instances.
+     *
+     * @return array
+     */
+    protected function reorder_main_exhibitions( $items ) {
+
+        // Return original $items array if search or year filter is used
+        if ( self::get_search_query_var() || self::get_year_query_var() ) {
+            return $items;
+        }
+
+        // Return null if there are no items in array
+        if ( $items === false ) {
+            return;
+        }
+
+        $items  = array_values( $items ); // reset array keys to start from 0
+        $length = count( $items );
+
+        // Loop through exhibitions and get main exhibitions to an array
+        for ( $i = 0; $i < $length; $i++ ) {
```

Comment:
Tässä vois tehdä tuon foreach ( $items as $key => $item ) {
}

Tää for käyttö lisää kompleksisuutta tähän